src/node.ts

```diff
@@ -97,10 +104,15 @@ export class EthereumNode extends constructs.Construct {
     // will be populated with defaults when passed to the node constructor
     this.network = props.network ?? Network.MAINNET;
     this.instanceType = props.instanceType ?? InstanceType.BURSTABLE3_LARGE;
-    this.availabilityZone = props.availabilityZone ?? `${region}a`;
 
-    utilities.validateRegion(region);
-    utilities.validateAvailabilityZone(region, this.availabilityZone);
+    // If no availability zone is provided, use the first in the region.
+    if (regionInEnvironment) {
+      utilities.validateRegion(region);
+      this.availabilityZone = props.availabilityZone ?? `${region}a`;
+      utilities.validateAvailabilityZone(region, this.availabilityZone);
+    } else {
+      this.availabilityZone = props.availabilityZone ?? availabilityZones[0];
```

Comment:
And again: `this.region = regions[0];` (I think this will work?)

Reply:
I set the region once before evaluating the if statement instead of in both conditionals because either way, the region will either be assigned a single token or a single string.